master/internal/command/shell_manager.go

```diff
@@ -72,40 +131,16 @@ func (s *shellManager) handleAPIRequest(ctx *actor.Context, apiCtx echo.Context)
 			respondBadRequest(ctx, err)
 			return
 		}
-
-		req, err := parseCommandRequest(apiCtx, s.db, &params, &s.taskSpec.TaskContainerDefaults)
-		if err != nil {
-			respondBadRequest(ctx, err)
-			return
-		}
-
-		if req.AgentUserGroup == nil {
-			req.AgentUserGroup = &s.defaultAgentUserGroup
-		}
-
-		var passphrase *string
-		if pwd, ok := req.Data["passphrase"]; ok {
-			if typed, typedOK := pwd.(string); typedOK {
-				passphrase = &typed
-			}
+		user := apiCtx.(*requestContext.DetContext).MustGetUser()
+		req := ShellLaunchRequest{
+			User:          &user,
+			CommandParams: &params,
 		}
-		generatedKeys, err := ssh.GenerateKey(passphrase)
+		summary, err := s.processShellLaunchRequest(ctx, req)
 		if err != nil {
-			ctx.Respond(err)
-			return
-		}
-
-		ctx.Log().Info("creating shell")
-
-		shell := s.newShell(req, generatedKeys)
-		if err := check.Validate(shell.config); err != nil {
 			respondBadRequest(ctx, err)
```

Comment:
For the old API, I think we've lost context on what is a "bad request" error and what is just a 500. I know the old API is deprecated essentially, but so long as the CLI and WebUI use it this could be misleading. Maybe `processShellLaunchRequest` needs an `ok bool` return to show it the request was ok or not?

Reply:
it I updated it so it returns a status code now.

Reply:
also good catch!